### ISSUE TITLE
Updated Feed Issue

### ISSUE THREAD

**Issue by hackenfu** [eric-pierce/freshapi#8]
Firstly, sorry for my English.
Some NEWS websites (for example WSJ) usually update their articles and as a result the feeds are updated too, but the feed IDs remain. So there is an updated feed, not a new feed.
In tt-rss, the list can change (title, time, ...) while some feeds update.
But in clients (https://github.com/eric-pierce/freshapi#compatible-clients, I have tried all iOS app except lire, both freshrss api & google api if it supports), no change happens.
I have no idea it's the API's issue, or the app's issue.
Sorry for disturb.

**Comment by eric-pierce** [comment]
@hackenfu thanks for the issue! I'll take a look and see if I can determine if it's an issue with the API implementation or just how TT-RSS handles updates.

**Comment by hackenfu** [comment]
> @hackenfu thanks for the issue! I'll take a look and see if I can determine if it's an issue with the API implementation or just how TT-RSS handles updates.

and i report two apps issue:
1. Readkit (ios): cannot add freshrss API. tiny tiny rss API works.
2. fluent reader (ios): can add google API, but there's no article.

**Comment by eric-pierce** [comment]
> > @hackenfu thanks for the issue! I'll take a look and see if I can determine if it's an issue with the API implementation or just how TT-RSS handles updates.
> 
> 
> 
> and i report two apps issue:
> 
> 1. Readkit (ios): cannot add freshrss API. tiny tiny rss API works.
> 
> 2. fluent reader (ios): can add google API, but there's no article.

I'll check these out as well. I didn't test apps that had native Tiny Tiny RSS support, but no reason not to improve support.

I'm out of town at a wedding this weekend so I won't be able to test until next week, but rest assured I'll check it out 😊

**Comment by hackenfu** [comment]
> @hackenfu thanks for the issue! I'll take a look and see if I can determine if it's an issue with the API implementation or just how TT-RSS handles updates.

Hello, I have done more tests. In those apps which support native tt-rss API (Fiery Feeds, ReadKit), it shows just the same as tt-rss show. tt-rss feed updates (title, time, ...), the app updates too. But freshapi DOESN'T.
looking forward to your test:)

**Comment by Tumetsu** [comment]
I noticed apparently similar problem with Linux [news-flash desktop application](https://github.com/patchedsoul/news-flash). I can log in and the desktop app pulls categories but it doesn't pull articles itself. At first I thought it might be reader issue but the same seemed to happen with also [Fluent Reader](https://hyliu.me/fluent-reader/) application. However, Liferea Google Reader API integration seems to work as well as Net News Wire on my iPhone.

**Comment by eric-pierce** [comment]
@Tumetsu interesting - I'll test the News Flash desktop app, and re-test Fluent Reader. Fluent Reader on iOS has had no issues for me. I'm betting you're using the desktop Fluent Reader app on Linux, is that right?

**Comment by Tumetsu** [comment]
Yes, I tried it briefly on desktop when I tried to figure out where the problem is . However, Newsflash is the one I’d prefer to use.

**Comment by eric-pierce** [comment]
Thanks @Tumetsu  and @hackenfu for your input. I found the issue impacting Readkit, Fluent Reader, and NewsFlash from populating articles, and it was fixed via https://github.com/eric-pierce/freshapi/issues/10

Taking a look at the original issue reported here (updated articles) as well now, and I'll comment here when I have some results.

**Comment by Tumetsu** [comment]
@eric-pierce Thank you! Confirmed it works on Newsflash on my desktop 👍

**Comment by w00dst0ck** [comment]
Hi! I currently have the same problem.

I have TTRSS as an installation on an Apache web server without Docker and so far everything works as desired with the Fever API and Reeder Classic.

In order to be able to use the extended features of the FreshAPI, I installed the plugin in the latest version and was able to test the function as specified in the documentation with “foo@bar:~$ curl -X POST --header ‘Authorization: GoogleLogin auth=yourusername/r4ih6gt412opqh11gptp3hodd6’ https://example.com/tt-rss/plugins.local/freshapi/api/greader.php/reader/api/0/subscription/export”.

In Reeder Classic I can set up access as FreshRSS and see the subscribed feeds incluive categories. 
However, no articles are synchronized.

Fluent Reader version 1.1.4 Windows shows the same behavior.

If I can contribute anything to the analysis, please let me know.

**Comment by eric-pierce** [comment]
@w00dst0ck interesting, I personally use Reeder Classic as my primary app and regularly test Fluent Reader and other apps and haven't run into this issue - in the error section of Tiny Tiny RSS do you see any errors?

I expect the issue has to do with your Apache - non-docker setup, but would love to help you solve. I'm betting it's an issue with Apache's PATH_INFO support on your configuration, do you know if your Apache install supports PATH_INFO requests?

If you're unsure, and are able to post your Apache configuration I'll take a look.

**Comment by w00dst0ck** [comment]
@eric-pierce because I'm on a shared webserver, l didn't have access to the config.

But I've added 

```
#Enable PATH_INFO
AcceptPathInfo On
```

to the .htaccess file to enable it.

Do you know if there is an easy way to verify if PATH_INFO is working as expected?

Thanks for the help!

**Comment by eric-pierce** [comment]
@w00dst0ck love it! The docker approach definitely keeps things simple and working - I haven't run into anyone else trying to use Apache with TT-RSS (the docker version uses nginx), but I'm sure there are some Apache specific issues both the application and FreshAPI would run into. Glad you're up and running!

**Comment by eric-pierce** [comment]
> Firstly, sorry for my English. Some NEWS websites (for example WSJ) usually update their articles and as a result the feeds are updated too, but the feed IDs remain. So there is an updated feed, not a new feed. In tt-rss, the list can change (title, time, ...) while some feeds update. But in clients (https://github.com/eric-pierce/freshapi#compatible-clients, I have tried all iOS app except lire, both freshrss api & google api if it supports), no change happens. I have no idea it's the API's issue, or the app's issue. Sorry for disturb.

@hackenfu This issue got a little hijacked with unrelated tasks, but I do want to circle back to the original issue you reported. I haven't been able to re-create the "not updating" issue you reported, do you have any examples you can highlight? I have set up a WSJ subscription on my test server so if you have an example ready I should be able to find the relevant article and test an update.

**Comment by hackenfu** [comment]
> > Firstly, sorry for my English. Some NEWS websites (for example WSJ) usually update their articles and as a result the feeds are updated too, but the feed IDs remain. So there is an updated feed, not a new feed. In tt-rss, the list can change (title, time, ...) while some feeds update. But in clients (https://github.com/eric-pierce/freshapi#compatible-clients, I have tried all iOS app except lire, both freshrss api & google api if it supports), no change happens. I have no idea it's the API's issue, or the app's issue. Sorry for disturb.
> 
> [@hackenfu](https://github.com/hackenfu) This issue got a little hijacked with unrelated tasks, but I do want to circle back to the original issue you reported. I haven't been able to re-create the "not updating" issue you reported, do you have any examples you can highlight? I have set up a WSJ subscription on my test server so if you have an example ready I should be able to find the relevant article and test an update.

Thank you for your continued follow-up. Taking Fiery Feeds as an example, when I subscribe to Tiny Tiny RSS, I use the official TT-RSS API. As long as there are updates to WSJ articles, the RSS feed will update as well. After Fiery Feeds syncs, it will be completely in sync with TT-RSS. However, if I subscribe to FreshRSS, using your API, no matter how Fiery Feeds syncs, as long as the article has appeared before, it will not change again (including the article title and update time). A simple way to reproduce this is by subscribing to WSJ via Fiery Feeds, waiting for 24 hours, and then comparing the RSS article list to see if there are any differences, since WSJ usually updates its previously published articles daily.
